src/packages/bem/bem.js

```diff
@@ -99,20 +99,25 @@ function bemStateless(blockDecl) {
                 classnamesMap: classnames,
             });
         },
-        elem: (elemName, props, extraMods = {}) => {
+        elem: (elemName, props, classesToKeep = '') => {
             if (!elemName || !props) {
                 throw new TypeError(
                     'elem(elemName, props) should be called with elem name and props as an arguments'
                 );
             }
-            return buildBemProps({
-                block: name,
-                elem: elemName,
-                props,
-                propsToMods,
-                extraMods,
-                classnamesMap: classnames,
-            });
+            const classes = [
+                buildBemProps({
+                    block: name,
+                    elem: elemName,
+                    props,
+                    propsToMods,
+                    extraMods: {},
+                    classnamesMap: classnames,
+                }).className,
+                classesToKeep,
+            ].filter(c => c);
+
+            return classes.length ? { className: classes.join(' ') } : null;
```

Comment:
Now return values of `block` and `elem` functions are inconsistent:

* `block` always return `{ className: '...' }`
* `elem` can return `null` or `{ className: '...' }`

I'd suggest that both functions return an object and if `className` is empty, then it's an empty object (so it can be safely spreaded).

Assuming this, it's more logical to move this logic into `buildBemProps` function from here.

Reply:
all this logic is now in benUtils